### PR TITLE
libtiff: update patches

### LINKED ISSUE #33858: libtiff patch urls out of date
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)
- What happened (include command output)
- What you expected to happen
- Step-by-step reproduction instructions (by running `brew install` commands)

```
$ brew install --build-from-source libtiff
==> Downloading https://download.osgeo.org/libtiff/tiff-4.0.9.tar.gz
Already downloaded: /Users/pxia/Library/Caches/Homebrew/downloads/8fa1037b01e387f472496f80e44b7ecce9fd1aed0744b6fec91c3c526ffd91ff--tiff-4.0.9.tar.gz
==> Downloading https://mirrors.ocf.berkeley.edu/debian/pool/main/t/tiff/tiff_4.0.9-6.de

curl: (22) The requested URL returned error: 404 Not Found
Trying a mirror...
==> Downloading https://mirrorservice.org/sites/ftp.debian.org/debian/pool/main/t/tiff/t

curl: (22) The requested URL returned error: 404 Not Found
Error: An exception occurred within a child process:
  DownloadError: Failed to download resource "libtiff--patch"
Download failed: https://mirrorservice.org/sites/ftp.debian.org/debian/pool/main/t/tiff/tiff_4.0.9-6.debian.tar.xz
```

### PR DESCRIPTION
**Pull request by fxcoudert**
libtiff build from source is broken (https://github.com/Homebrew/homebrew-core/issues/33858) because Debian has remove the file we used for their patches. And their are now building from a git checkout, without patches.

So this PR updates our `libtiff` formula with patches from the latest Fedora package.

Fixes #33858

@Homebrew/core review welcome

### CODE REVIEW

File: Formula/libtiff.rb [hunk 1]
```diff
@@ -21,19 +21,57 @@ class Libtiff < Formula
 
   # All of these have been reported upstream & should
   # be fixed in the next release, but please check.
+  # Patches are taken from latest Fedora package, which is currently
+  # libtiff-4.0.9-13.fc30.src.rpm
```
Comment (claui) [inline Formula/libtiff.rb]:
```suggestion
  # libtiff-4.0.9-13.fc30.src.rpm and whose changelog can be found at:
  # https://apps.fedoraproject.org/packages/libtiff/changelog/
```

It took me a long time to find the changelog for Fedora’s `fc30` revision; they didn’t turn up at several Google searches. The revision is also [not even included in Fedora’s Git repo](https://src.fedoraproject.org/cgit/rpms/libtiff.git) yet. So I feel the full URL of the changelog should be included via comment.

File: Formula/libtiff.rb [hunk 2]
```diff
@@ -21,19 +21,57 @@ class Libtiff < Formula
...
-          "patches/CVE-2018-5784.patch",
-          "patches/CVE-2017-11613_part1.patch",
-          "patches/CVE-2017-11613_part2.patch",
-          "patches/CVE-2018-7456.patch",
-          "patches/CVE-2017-17095.patch",
-          "patches/CVE-2018-8905.patch",
-          "patches/CVE-2018-10963.patch"
+    url "https://raw.githubusercontent.com/Homebrew/formula-patches/899be6424defc28747be318e28d57a4bf9bcc7c6/libtiff/libtiff-CVE-2018-8905.patch"
+    sha256 "de9630eb6ee7cfff4643c6de3d1c47a7ced5aef4c4ac4f74dc40b5211e814e5d"
   end
```
Comment (claui) [inline Formula/libtiff.rb]: There has been a recent security issue with libtiff \([CVE-2018-18557](https://security-tracker.debian.org/tracker/CVE-2018-18557)\). The new Git snapshot used by Debian [includes a fix](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=905798) for the CVE. However, Fedora [hasn’t issued a new release yet](https://apps.fedoraproject.org/packages/libtiff/changelog/) that includes the fix.

If we absolutely don’t want to base the formula on the Git snapshot, should we not find another way to obtain a patch and include it in the formula?

**Review by claui — CHANGES_REQUESTED** [review]

**Comment by fxcoudert** [comment]
git snapshots are definitely against our policy. It's sad that libtiff is not releasing regularly, and that Debian is enabling them. Patch added for CVE-2018-18557

**Review by claui — APPROVED** [review]
Thanks @fxcoudert, and also 👍 for cherry-picking the diff from upstream directly.

**Comment by David070** [comment]
Hey I am quite new to Python and Libtiff.. I tried to install it via Brewmaster but its giving me Error_messages probably due to the deletion of the file?!
Is there another way to install it? Can you maybe help me.. 
Thank you, David :)

**Comment by claui** [comment]
@David070 Care to re-post your question on [discourse.brew.sh](https://discourse.brew.sh/) in a bit more detail? This makes sure your support request will be read by the right people, and can be acted upon. Thanks!

**Comment by claui** [comment]
@fxcoudert It does now 😉

**Comment by fxcoudert** [comment]
@claui I like the idea, but it looks a bit hackish… maybe we can adapt the patch and host it on `formula-patches`, what do you think?

**Comment by fxcoudert** [comment]
Thanks @claui for the help!

**Pull request closed without merging**